lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -0,0 +1,1068 @@
+			Object.assign(object, {
+				name,
+				chunks: chunkGroup.chunks.map(c => c.id),
+				assets: chunkGroup.chunks.reduce(
+					(array, c) => array.concat(c.files || []),
+					/** @type {string[]} */ ([])
+				),
+				children: Object.keys(children).reduce((obj, key) => {
+					const groups = children[key];
+					obj[key] = groups.map(group => ({
+						name: group.name,
+						chunks: group.chunks.map(c => c.id),
+						assets: group.chunks.reduce(
+							(array, c) => array.concat(c.files || []),
+							/** @type {string[]} */ ([])
+						)
+					}));
+					return obj;
+				}, /** @type {Record<string, {name: string, chunks: (string|number)[], assets: string[]}[]>} */ Object.create(null)),
+				childAssets: Object.keys(children).reduce((obj, key) => {
+					const groups = children[key];
+					obj[key] = Array.from(
+						groups.reduce((set, group) => {
+							for (const chunk of group.chunks) {
+								for (const asset of chunk.files) {
+									set.add(asset);
+								}
+							}
+							return set;
+						}, /** @type {Set<string>} */ (new Set()))
+					);
+					return obj;
+				}, Object.create(null))
+			});
+		},
+		performance: (object, { chunkGroup }) => {
+			object.isOverSizeLimit = SizeLimitsPlugin.isOverSizeLimit(chunkGroup);
+		}
+	},
+	module: {
+		_: (object, module, context, { requestShortener }, factory) => {
+			const { compilation, type } = context;
+			const { chunkGraph, moduleGraph } = compilation;
+			/** @type {Module[]} */
+			const path = [];
+			const issuer = moduleGraph.getIssuer(module);
+			let current = issuer;
+			while (current) {
+				path.push(current);
+				current = moduleGraph.getIssuer(current);
+			}
+			path.reverse();
+			const profile = moduleGraph.getProfile(module);
+			Object.assign(object, {
+				id: chunkGraph.getModuleId(module),
+				identifier: module.identifier(),
+				name: module.readableIdentifier(requestShortener),
+				index: moduleGraph.getPreOrderIndex(module),
+				preOrderIndex: moduleGraph.getPreOrderIndex(module),
+				index2: moduleGraph.getPostOrderIndex(module),
+				postOrderIndex: moduleGraph.getPostOrderIndex(module),
+				size: module.size(),
+				sizes: Array.from(module.getSourceTypes()).reduce((obj, type) => {
+					obj[type] = module.size(type);
+					return obj;
+				}, {}),
+				cacheable: module.buildInfo.cacheable,
+				built: compilation.builtModules.has(module),
+				optional: module.isOptional(moduleGraph),
+				runtime: module.type === "runtime",
+				chunks: Array.from(
+					chunkGraph.getOrderedModuleChunksIterable(module, compareChunksById),
+					chunk => chunk.id
+				),
+				issuer: issuer && issuer.identifier(),
+				issuerId: issuer && chunkGraph.getModuleId(issuer),
+				issuerName: issuer && issuer.readableIdentifier(requestShortener),
+				issuerPath:
+					issuer && factory.create(`${type}.issuerPath`, path, context),
+				failed: module.errors && module.errors.length > 0,
```

Comment:
In order to ensure type safety, I think the following would be better:
```suggestion
 				failed: module.errors ? module.errors.length > 0 : false,
```